src/codegen/build_module.cc

```diff
@@ -572,6 +608,54 @@ void GenericFunc::CallPacked(TVMArgs args, TVMRetValue* ret) const {
   func.CallPacked(args, ret);
 }
 
+TVM_REGISTER_API("_GetCurrentBuildConfig")
+.set_body([](TVMArgs args, TVMRetValue* ret) {
+  *ret = BuildConfig::current_build_config();
+  });
+
+TVM_REGISTER_API("_EnterBuildConfigScope")
+.set_body([](TVMArgs args, TVMRetValue* ret) {
+  BuildConfig target = args[0];
+  BuildConfig::EnterBuildConfigScope(target);
+  });
+
+TVM_REGISTER_API("_ExitBuildConfigScope")
+.set_body([](TVMArgs args, TVMRetValue* ret) {
+  BuildConfig::ExitBuildConfigScope();
+  });
+
+TVM_REGISTER_API("_BuildConfigSetAddLowerPass")
+.set_body([](TVMArgs args, TVMRetValue* ret) {
+  BuildConfig cfg = args[0];
+  std::vector<std::tuple<int, PackedFunc>> add_lower_pass;
+  CHECK_EQ(args.size() % 2, 1);
+  for (int i = 1; i < args.size(); i += 2) {
+    add_lower_pass.push_back(std::make_tuple(
+      args[i].operator int(),
+      args[i + 1].operator tvm::runtime::PackedFunc()));
+  }
+  cfg.set_add_lower_pass(add_lower_pass);
+  });
+
+TVM_REGISTER_API("_BuildConfigGetAddLowerPassSize")
```

Comment:
Just have one function, that dispatches on arg[0] (type) and returns size, item and phase